Tests/Marketplace/search_and_install_packs.py

```diff
@@ -434,46 +369,57 @@ def search_pack_and_its_dependencies(client: demisto_client,
             Each list contain one pack and its dependencies.
 
     """
-    pack_data = {}
-    if pack_id not in packs_to_install:
-        pack_display_name = get_pack_display_name(pack_id)
-        if pack_display_name:
-            pack_data = search_pack(client, pack_display_name, pack_id, lock)
-        if pack_data is None:
-            pack_data = {
-                'id': pack_id,
-                'version': '1.0.0'
-            }
-
-    if pack_data:
-        dependencies = get_pack_dependencies(client, pack_data, lock)
-
-        current_packs_to_install = [pack_data]
-        if dependencies:
-            # Check that the dependencies don't include a deprecated pack:
-            for dependency in dependencies:
-                pack_path = os.path.join(PACKS_FOLDER, dependency.get('id'))
-                if is_pack_deprecated(pack_path):
-                    logging.critical(f'Pack {pack_id} depends on pack {dependency.get("id")} which is a deprecated '
-                                     f'pack.')
-                    global SUCCESS_FLAG
-                    SUCCESS_FLAG = False
-                else:
-                    current_packs_to_install.append(dependency)
-
-        lock.acquire()
-        if one_pack_and_its_dependencies_in_batch:
-            pack_and_its_dependencies = \
-                {p['id']: p for p in current_packs_to_install if p['id'] not in packs_in_the_list_to_install}
-            if pack_and_its_dependencies:
-                packs_in_the_list_to_install += pack_and_its_dependencies
-                batch_packs_install_request_body.append(list(pack_and_its_dependencies.values()))  # type:ignore[union-attr]
-        else:
-            for pack in current_packs_to_install:
-                if pack['id'] not in packs_to_install:
-                    packs_to_install.append(pack['id'])
-                    installation_request_body.append(pack)
-        lock.release()
+    api_data = get_pack_dependencies(client, pack_id, lock)
+
+    if not api_data:
+        return
+
+    pack_data = api_data["packs"][0]
```

Comment:
Because as far as I know, a successful response for this endpoint should always have the same structure, and I think it would be better to fail on an error so that we'll know something is wrong, than treating it gracefully as an empty list and continuing.